src/CodeGeneration/ApiGenerator/Domain/ApiEndpoint.cs

```diff
@@ -265,12 +265,38 @@ public IEnumerable<CsharpMethod> CsharpMethods
 							Url = this.Url
 						};
 						PatchMethod(apiMethod);
+						_csharpMethods.Add(apiMethod);
 						yield return apiMethod;
 					}
 				}
 			}
 		}
 
+		private void PatchEndpoint()
+		{
+			//rename the {metric} route param to something more specific on XpackWatcherStats
+			// TODO: find a better place to do this
+			if (this.CsharpMethodName == "XpackWatcherStats")
+			{
+				var metric = this.Url.Parts.First(p => p.Key == "metric");
+
+				var apiUrlPart = metric.Value;
+				apiUrlPart.Name = "watcher_stats_metric";
```

Comment:
The renaming is really for our code generation and enum naming purposes, but if it would help others, I will.